test/v2/USTBv2TokenStorageLayoutTests.t.sol

```diff
@@ -0,0 +1,162 @@
+            assertEq(SuperstateToken(address(currentToken)).owner(), expectedOwner);
+
+            // assert _pendingOwner (storage slot 201)
+            address pendingOwnerSlotValue = address(uint160(uint256(loadSlot(201))));
+            address expectedPendingOwner = address(0x0);
+            assertEq(pendingOwnerSlotValue, expectedPendingOwner);
+
+            // assert _pendingOwner from contract methods
+            assertEq(SuperstateToken(address(currentToken)).pendingOwner(), expectedPendingOwner);
+        } else {
+            // V1 did not support this field, and so ignoring
+        }
+    }
+
+    // See note in `SuperstateTokenStorageLayoutTestBase` to see why this is being overwritten
+    function assertSuperstateTokenStorageLayout(bool hasUpgraded) public override {
+        if (hasUpgraded) {
+            // V2 storage layout, ignore encumberances/nonces/accountingPaused as they are corrupted (this is known and accepted)
+        } else {
+            // V1 storage layout
+
+            // assert nonces (stored in storage slot 151)
+            bytes32 noncesSlot = keccak256(abi.encode(eve, uint256(151)));
+            uint256 noncesSlotValue = uint256(vm.load(address(tokenProxy), noncesSlot));
+            uint256 expectedNonce = 1;
+            assertEq(noncesSlotValue, expectedNonce);
+
+            // assert nonces from contract method
+            assertEq(currentToken.nonces(eve), 1);
+
+            // assert encumberedBalanceOf (stored in storage slot 152)
+            bytes32 encumberedBalanceOfSlot = keccak256(abi.encode(alice, uint256(152)));
+            uint256 encumberedBalanceOfSlotValue = uint256(vm.load(address(tokenProxy), encumberedBalanceOfSlot));
+            uint256 expectedAliceEncumberedBalanceOf = 5e6;
+            assertEq(encumberedBalanceOfSlotValue, expectedAliceEncumberedBalanceOf);
+
+            // assert encumberedBalanceOf from contract method
+            assertEq(currentToken.encumberedBalanceOf(alice), expectedAliceEncumberedBalanceOf);
+
+            // assert encumbrances (stored in storage slot 153).
+            bytes32 encumberancesOwnerSlot = keccak256(abi.encode(alice, uint256(153)));
+            bytes32 encumberancesTakerSlot = keccak256(abi.encode(bob, uint256(encumberancesOwnerSlot)));
+            uint256 encumberancesTakerSlotValue = uint256(vm.load(address(tokenProxy), encumberancesTakerSlot));
+            assertEq(encumberancesTakerSlotValue, expectedAliceEncumberedBalanceOf);
+
+            // assert encumbrances from contract method
+            assertEq(currentToken.encumbrances(alice, bob), expectedAliceEncumberedBalanceOf);
+
+            // assert accountingPaused (storage slot 154)
+            uint256 accountingPausedSlotValue = uint256(loadSlot(154));
+            assertEq(1, accountingPausedSlotValue);
+
+            // assert accountingPaused from contract method
+            assertEq(true, currentToken.accountingPaused());
+
+            // assert decimals
```

Comment:
RM or TODO?

Reply:
Replaced with `// ignore decimals due to being in-lined in bytecode`